src/event.cc

```diff
@@ -88,27 +91,25 @@ EventDialer::print(std::ostream &os) const
     os << ')';
 }
 
-ev_entry::ev_entry(char const * aName, EVH * aFunction, void * aArgument, double evWhen, int aWeight, bool haveArg) :
-    name(aName),
-    func(aFunction),
-    arg(haveArg ? cbdataReference(aArgument) : aArgument),
+ev_entry::ev_entry(double evWhen, int aWeight, const AsyncCall::Pointer &aCall) :
     when(evWhen),
     weight(aWeight),
-    cbdata(haveArg),
-    next(NULL)
+    call(aCall)
 {
 }
 
-ev_entry::~ev_entry()
+static std::ostream &
+operator <<(std::ostream &os, ev_entry &e)
 {
-    if (cbdata)
-        cbdataReferenceDone(arg);
+    os << e.call->name << " [" << e.call->id << ']';
+    return os;
 }
 
-void
-eventAdd(const char *name, EVH * func, void *arg, double when, int weight, bool cbdata)
+EventScheduler &
+Events()
 {
-    EventScheduler::GetInstance()->schedule(name, func, arg, when, weight, cbdata);
+    static EventScheduler instance;
+    return instance;
```

Comment:
If you insist on renaming EventScheduler::GetInstance() in this PR, then please fix the implementation so that the instance remains alive throughout program termination, when some class destructors might call it to cancel scheduled events.

```suggestion
    static const auto instance = new EventScheduler();
    return *instance;
```

Reply:
> If you insist on renaming EventScheduler::GetInstance() in this PR, 

I did not insist on that. You complained that GetInstance should not exist.